### PR TITLE
Improve documentation about redirects

### PR DESCRIPTION
**Pull request by DaazKu**
Fixes https://github.com/vanilla/docs/issues/34

### CODE REVIEW

File: content/developer/importing/porter.md [hunk 1]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
 
 #### Redirects
 
-Many legacy platform redirects can be handled by the **Redirector** plugin in the [Addons repository](https://github.com/vanilla/addons). Simply enable the plugin. Check the description for what it supports currently.
+For everything to work properly you will probably need to do 2 redirects!
+One redirect for the [path](#redirect-the-path) and one for the [domain](#redirect-the-domain).
```
Comment (initvector) [inline content/developer/importing/porter.md]: > For everything to work properly you will probably need to do two redirects: one for the [path](#redirect-the-path) and one for the [domain](#redirect-the-domain).

File: content/developer/importing/porter.md [hunk 2]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
...
+When setup properly, users hitting your old forum should then be
+1) globally 301 redirected to your new domain hosting Vanilla's forum which will then
+2) 301 redirect to the correct path based on the URL requested.
 
-If you run the import on localhost and then copy the database to production, be sure to manually transfer those special Routes as well.
+##### Redirect the path
 
-Once you have correct redirects set in Vanilla, you must set up a global redirect from your old forum address to Vanilla. See the `forum-redirector` folder in the [Addons repository](https://github.com/vanilla/addons) for help setting this up. This is *not* the same as the Redirector plugin above (which is at `plugins/Redirector`).
+The "path" is everything that comes after the domain.
+In `http://domain.com/forum/thread.php?id=10&pageNumber=2` that would be `/forum/thread.php?id=10&pageNumber=2`
```
Comment (initvector) [inline content/developer/importing/porter.md]: >The "path" is everything that comes after the domain. In `http://domain.com/forum/thread.php?id=10&pageNumber=2` that would be `/forum/thread.php?id=10&pageNumber=2`

File: content/developer/importing/porter.md [hunk 3]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
...
 
-Users hitting your old forum should then be 1) globally 301 redirected to Vanilla which will then 2) 301 redirect to the correct content based on the URL requested. Yes, 2 redirects are used in this technique.
+Many legacy platform path redirects can be handled by the **Redirector** plugin (not the same thing than the `forum-redirector` folder) in the [Addons repository](https://github.com/vanilla/addons).
+Simply enable the plugin. Check the description for what it supports currently.
```
Comment (initvector) [inline content/developer/importing/porter.md]: > Many legacy platform path redirects can be handled by the **Redirector** plugin in the [Addons repository](https://github.com/vanilla/addons).  Simply install and enable the plugin. You may check the description for what it platforms and redirects it currently supports. Forum Redirector, which is under "forum-redirector" in the same repo, is not the same thing and should not be used for these purposes.

File: content/developer/importing/porter.md [hunk 4]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
...
 
+You can also create custom redirects as needed using the [Routes](/developer/routes) feature in the Dashboard.
+Use regular expressions to match incoming URL patterns and 301 redirects to their new place.
```
Comment (initvector) [inline content/developer/importing/porter.md]: > You may also create custom redirects as needed using the [Routes](/developer/routes) feature in the Dashboard.  Redirects created here can be temporary (302) or permanent (301).  Regular expressions can be used to perform pattern matching against incoming URLs.

File: content/developer/importing/porter.md [hunk 5]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
...
 
-## Troubleshooting
+An example of a path redirect would be:
+`/forum/discussion.php?id=10&pageNumber=2`
+being redirected to
+`/discussion/10?page=2`
+which is the correct path for the discussion number 10 on page 2.
```
Comment (initvector) [inline content/developer/importing/porter.md]: > An example of a path redirect would be `/forum/discussion.php?id=10&pageNumber=2` being redirected to `/discussion/10?page=2`, which is the correct path for the discussion number 10 on page 2.

File: content/developer/importing/porter.md [hunk 6]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
...
+Example:
+If your new forum reside on a different domain you will need to change the domain name like so
+`http://olddomain.com/forum/discussion.php?id=10&page=2` to `http://newdomain.com/forum/discussion.php?id=10&page=2`
+This should result in `http://newdomain.com/forum/discussion.php?id=10&page=2` being redirected to `http://newdomain.com/discussion/10?page=2`.
+
+*Note*: For the pages to work properly you need to configure Vanilla to have the same number of comments per page than your old forum had.
 
+##### Redirect the domain
+
+If the domain where your new forum reside is different than where your old forum was you will need to redirect the domain.
```
Comment (initvector) [inline content/developer/importing/porter.md]: > ...new forum resides...

File: content/developer/importing/porter.md [hunk 7]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
...
+
+Once you setup the paths redirects you should test them thoroughly to make sure that everything works fine.
+To test you can simple copy an URL from your old forum to the new one.
+
+Example:
+If your new forum reside on a different domain you will need to change the domain name like so
+`http://olddomain.com/forum/discussion.php?id=10&page=2` to `http://newdomain.com/forum/discussion.php?id=10&page=2`
+This should result in `http://newdomain.com/forum/discussion.php?id=10&page=2` being redirected to `http://newdomain.com/discussion/10?page=2`.
+
+*Note*: For the pages to work properly you need to configure Vanilla to have the same number of comments per page than your old forum had.
```
Comment (initvector) [inline content/developer/importing/porter.md]: > ... comments per page as your old forum had.

File: content/developer/importing/porter.md [hunk 8]
```diff
@@ -141,37 +141,84 @@ If you run into problems, your platform isn't supported, or want to reset them a
...
+- Discussions / Threads
+- Comments / Replies
+- User profiles
+
+To summarize, all public pages **with relevant content** that were ported to Vanilla *should* be redirected.
+
+##### Testing the redirects
+
+Once you setup the paths redirects you should test them thoroughly to make sure that everything works fine.
+To test you can simple copy an URL from your old forum to the new one.
```
Comment (initvector) [inline content/developer/importing/porter.md]: > Once you setup the path redirects you should test them thoroughly to make sure that everything works fine. To test, you can simple copy a URL from your old forum to the new one.

File: content/developer/importing/porter.md (hunk 2 above)
Comment (DaazKu) [inline content/developer/importing/porter.md]: You realize that this line break is only visible in the source?
It is much easier to read these, when disposed like so, in the editor.

**Pull request merged**